src/audio_generic.cpp

```diff
@@ -22,31 +22,20 @@
 #include <memory>
 #include "audio_decoder_midi.h"
 #include "audio_generic.h"
-#include "audio_generic_midiout.h"
```

Comment:
7269d325035f547907ec352bff5d39bc611abc88 breaks compilation with for platforms libretro, psvita, sdl and switch when compiling audio.o because they include `audio_generic.h`. Since this commit, said header includes `audio_generic_midiout.h` which does not provide a definition for `AudioDecoderMidi` itself.
This used to be no problem because `audio_generic_midiout.h` was only included from `audio_generic.cpp` which also includes the missing header.

The fix is to include `audio_decoder_midi.h` either in `audio_generic.h` or in `audio_generic_midiout.h`.

Reply:
C++ Header mess :/. Thanks for testing this!